### ISSUE TITLE
Enum validation

### ISSUE THREAD

**Issue by marcelosousa** [redhat-developer/vscode-yaml#803]
## Describe the bug
I'm getting unexpected behaviour with the validation of the schema when it comes to enums.

For example, consider the following schema:

```
{
    "$schema": "http://json-schema.org/draft-07/schema",
    "definitions": {
        "rule": {
            "description": "A rule.",
            "type": "object",
            "properties": {
                "kind": {
                    "description": "The kind of a rule.",
                    "type": "string",
                    "enum": [
                        "tested"
                    ]
                }
            },
            "required": [
                "kind"
            ],
            "additionalProperties": false
        }
    },
    "properties": {
        "rules": {
            "description": "Rule list",
            "type": "array",
            "items": {
                "$ref": "#/definitions/rule"
            },
            "minProperties": 1,
            "additionalProperties": false
        }
    }
}
```

And the following YML file:

```
rules:
  - kind: test
```

## Expected Behavior
We should get an error in `test` as it is not an element of the enum.

## Current Behavior
No error. It seems that as long it is as prefix of an element in the enum, the validation succeeds.

## Environment
- [x] Mac

**Comment by msivasubramaniaan** [comment]
Issue has been resolved and it will be available on the next v1.10.0

**Comment by marcelosousa** [comment]
Many thanks @msivasubramaniaan 🙌
